include/nuttx/atomic.h

```diff
@@ -78,6 +78,9 @@ extern "C++"
 #    if !(__clang__) && defined(__cplusplus)
 #      define _Atomic
 #    endif
+#    if (__clang__)
+#      define _CLANG_DISABLE_CRT_DEPRECATION_WARNINGS
```

Comment:
let's use the new macro or remove ATOMIC_VAR_INIT from code base

Reply:
Done,I used ATOMIC_INIT instead of ATOMIC_VAR_INIT. Normally it is equivalent to ATOMIC_VAR_INIT. In clang it means ATOMIC_INIT(value) (value)

Reply:
ATOMIC_VAR_INIT is deprecated, so we should change `ATOMIC_VAR_INIT(x)` to `x`.